### PR TITLE
Update publishing-bot rules for Go 1.17.7 / 1.16.14

### PR DESCRIPTION
**Pull request by xmudrii**
#### What type of PR is this?

/kind feature
/area dependency

#### What this PR does / why we need it:

Update publishing-bot rules for Go 1.17.7 / 1.16.14.

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/release/issues/2425

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/assign @cpanato @saschagrunert @palnabarun @puerco @justaugustus
cc @kubernetes/release-engineering

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
until #108102, #108101, and #108100 are not merged

**Review by saschagrunert — APPROVED** [review]

**Comment by xmudrii** [comment]
/hold cancel
All branches are now updated

**Pull request merged**
